src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -0,0 +1,109 @@
+#ifndef RAY_CORE_WORKER_DIRECT_ACTOR_TRANSPORT_H
+#define RAY_CORE_WORKER_DIRECT_ACTOR_TRANSPORT_H
+
+#include <list>
+
+#include "ray/core_worker/transport/transport.h"
+#include "ray/raylet/raylet_client.h"
+#include  "ray/gcs/gcs_client.h"
+#include "ray/rpc/worker/direct_actor_client.h"
+#include "ray/rpc/worker/direct_actor_server.h"
+#include "ray/core_worker/object_interface.h"
+
+namespace ray {
+
+/// In raylet task submitter and receiver, a task is submitted to raylet, and possibly
+/// gets forwarded to another raylet on which node the task should be executed, and
+/// then a worker on that node gets this task and starts executing it.
+
+struct PendingTaskRequest {
+  TaskID task_id;
+  int num_returns;
+  std::unique_ptr<rpc::PushTaskRequest> request;
+};
+
+
+class CoreWorkerDirectActorTaskSubmitter : public CoreWorkerTaskSubmitter {
```

Comment:
I found `CoreWorkerTaskSubmitter`'s destructor isn't defined as virtual. Can you fix in this RP?

Reply:
sure, thanks for the reminder.